lib/shared/puppet_forge/module_name.rb

```diff
@@ -0,0 +1,23 @@
+module PuppetForge
+  # Normalize Forge module names
+  #
+  # Forge modules use '-' as the separator between the username and modulename
+  # in the module slug, but the Puppet Forge uses '/' as the separator on module
+  # pages and in the URL. This class hides the distinction by always using '-' as
+  # the separator for normal use. If the name needs to use the slash separated name
+  # then the #forge_name method can be used.
+  #
+  # @api private
+  class ModuleName < String
```

Comment:
I'm wondering if ModuleName should also be namespaced under V3, it seems unlikely but not impossible that a future Forge API might change the syntax of module names...